table_foosball/database_test.rb

```diff
@@ -0,0 +1,79 @@
+require_relative './database'
+require_relative './table_foosball_01'
+require 'minitest/autorun'
+
+class DatabaseTest < Minitest::Test
+	def setup
+		File.open("./games_test.sdb", "w") {}
+		File.open("./players_test.sdb", "w") {}
+
+		@gamedb = Database.new(:games_test)
+		@playerdb = Database.new(:players_test)
+
+		@sebastian = Player.new("Sebastian")
+		@simon = Player.new("Simon")
+		@daniel = Player.new("Daniel")
+		@kenichi = Player.new("Kenichi")
+
+		@game1 = Game.new([@sebastian, @simon, @daniel], [@kenichi], 10, 0)
+		@game2 = Game.new([@simon], [@daniel], 6, 10)
+		@game3 = Game.new([@sebastian, @simon], [@daniel, @kenichi], 10, 5)
+		@game4 = Game.new([@sebastian], [@daniel, @kenichi], 10, 9)
+
+		@players = [@sebastian, @simon, @daniel, @kenichi]
+		@games = [@game1, @game2, @game3, @game4]
+	end
+
+	def teardown
+		Game.games = nil
+		File.delete("./games_test.sdb", "./players_test.sdb")
+	end
+
+	def test_create_new_database
+		name = :test_database
+		test_database = Database.new(name)
+		
+		assert_equal Database, test_database.class
+		assert_equal name, test_database.instance_eval { @name }
+		assert_equal "./#{name}.sdb", test_database.instance_eval { @filename }
+	end
+
+	def test_save_and_load_games_on_database
+		assert_equal "", File.read("./games_test.sdb")
+		assert_empty @gamedb.load
+
+		@gamedb.save(@games)
+		
+		refute_equal "", File.read("./games_test.sdb")
+		compare_object_states @games, @gamedb.load
+	end
+
+	def test_save_and_load_players_on_database
+		assert_equal "", File.read("./players_test.sdb")
+		assert_empty @playerdb.load
+
+		@playerdb.save(@players)
+		
+		refute_equal "", File.read("./players_test.sdb")
+		compare_object_states @players, @playerdb.load
+	end
+
+	private
+
+	def compare_object_states exp, act
```

Comment:
I see what you're trying to do here. What you ran into is that two objects can't be compared cleanly, e.g. if I do:

```ruby
class Foo
  def initialize(bar)
    @bar = bar
  end
end
```

And then do:

```ruby
Foo.new("a") == Foo.new("a")
# => false
```

This is because the default `==` method on objects will only look for complete equality, but you could define your own.

I think your helper here is actually the most future-proof way of doing this, but if each object had an equality method, you wouldn't need this:


```ruby
class Foo
  attr_reader :bar

  def initialize(bar)
    @bar = bar
  end

  def ==(other)
    other.bar == self.bar
  end
end
```

In which case it'd work, e.g. `foo1 = Foo.new("a"); foo2 = Foo.new("a"); [foo1] == [foo2]` would equal true.